### PR TITLE
doc: twister: Describe hardware fixtures 

### PR DESCRIPTION
**Pull request by bene42**
Hi,
while working on https://github.com/zephyrproject-rtos/zephyr/pull/75287 I stumbled around the the extraordinary configuration and use of hardware related fixtures.
The lack of a central overview and description of hardware related fixtures cases first confusion and then acid regurgitation.
Add basic documentation to avoid some of my pain points.

Regards
Benedikt Spranger

### CODE REVIEW

**Review by nashif — APPROVED** [review]

**Comment by nashif** [comment]
@gchwier @golowanow @PerMac  @hakehuang  please review

File: scripts/gen_hw_fixture_table.sh [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+then
+    echo "usage: $0 <hardware-fixture-table>"
+    exit 1
+fi
+
+TOPDIR=$(git rev-parse --show-toplevel)
+
+find $TOPDIR -name testcase.yaml -exec awk '/fixture:/{print $2}' {} \; | \
+    sort -u | \
+    awk -voutfile="$1" '
```
Comment (golowanow) [inline scripts/gen_hw_fixture_table.sh:18]: When I try the script with, for example, one fixture removed from the doc file to imitate a new fixture appears, the script adds it with no description. It is a bit confusing. 
May it be a link to the testcase.yaml file(s) where the fixture appears, or even a cross-reference column here to point what test suites need the fixture ?

```
~/zephyrproject/zephyr$ git diff
diff --git a/doc/develop/test/twister/hardware-fixtures.rst b/doc/develop/test/twister/hardware-fixtures.rst
index 06325ed4798..d57b969417e 100644
--- a/doc/develop/test/twister/hardware-fixtures.rst
+++ b/doc/develop/test/twister/hardware-fixtures.rst
@@ -28,8 +28,6 @@ Hardware fixture table
 +----------------------------------+--------------------------------------------------------------------------------------------+
 | i2c_bus_short                    | The DUT has two I2C busses connected together and the I2C controller are DUAL role capable |
 +----------------------------------+--------------------------------------------------------------------------------------------+
-| i2c_loopback                     | The DUT has an I2C master controller connected to an I2C slave controller                  |
-+----------------------------------+--------------------------------------------------------------------------------------------+
 | i2c_ram                          | The DUT has a MB85 FeRAM connected to an I2C bus                                           |
 +----------------------------------+--------------------------------------------------------------------------------------------+
 | maxim_ds3231                     | The DUT is equipped with a Maxim DS3231 chip                                               |

~/zephyrproject/zephyr$ ./scripts/gen_hw_fixture_table.sh ./doc/develop/test/twister/hardware-fixtures.rst

~/zephyrproject/zephyr$ git diff
diff --git a/doc/develop/test/twister/hardware-fixtures.rst b/doc/develop/test/twister/hardware-fixtures.rst
index 06325ed4798..b2927355613 100644
--- a/doc/develop/test/twister/hardware-fixtures.rst
+++ b/doc/develop/test/twister/hardware-fixtures.rst
@@ -28,7 +28,7 @@ Hardware fixture table
 +----------------------------------+--------------------------------------------------------------------------------------------+
 | i2c_bus_short                    | The DUT has two I2C busses connected together and the I2C controller are DUAL role capable |
 +----------------------------------+--------------------------------------------------------------------------------------------+
-| i2c_loopback                     | The DUT has an I2C master controller connected to an I2C slave controller                  |
+| i2c_loopback                     |                                                                                            |
 +----------------------------------+--------------------------------------------------------------------------------------------+
 | i2c_ram                          | The DUT has a MB85 FeRAM connected to an I2C bus                                           |
 +----------------------------------+--------------------------------------------------------------------------------------------+
```

File: doc/develop/test/twister.rst [hunk 2]
```diff
@@ -988,6 +988,17 @@ The ``--device-flash-with-test`` option indicates that on the platform
 the flash operation also executes a test scenario, so the flash timeout is
 increased by a test scenario timeout.
 
+Special hardware preliminary
```
Comment (PerMac) [inline doc/develop/test/twister.rst:991]: "Special hardware preliminary" is this a correct English term? Does it mean "Additional hardware requirements"

File: scripts/gen_hw_fixture_table.sh [hunk 3]
```diff
@@ -0,0 +1,92 @@
+#! /bin/bash
+#
+# Copyright (c) 2024 Linutronix GmbH
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
```
Comment (PerMac) [inline scripts/gen_hw_fixture_table.sh:7]: Can you please add some comments about how to use this script and what it does?

**Review by PerMac — CHANGES_REQUESTED** [review]

File: doc/develop/test/twister/hardware-fixtures.rst
Comment (golowanow) [inline doc/develop/test/twister/hardware-fixtures.rst:1]: Please put a warning in this file with:
1) this file is autogenerated;
2) how to update this file properly.

**Review by golowanow — CHANGES_REQUESTED** [review]

File: doc/develop/test/twister.rst (hunk 2 above)
Comment (golowanow) [reply doc/develop/test/twister.rst:991]: The [`Fixtures`](https://github.com/zephyrproject-rtos/zephyr/blob/9d9089edd09919c90c4224222fc2c560410e6c85/doc/develop/test/twister.rst?plain=1#L1178) section below has some details already, so it looks better to add a link to the new page there.

File: doc/develop/test/twister/hardware-fixtures.rst [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+| dac_adc_loopback                 | The DUT has ADC and DAC connected                                                          |
++----------------------------------+--------------------------------------------------------------------------------------------+
+| external_flash_mx25l51245g       | The DUT has a mx25l51245g QSPI flash connected to the board                                |
++----------------------------------+--------------------------------------------------------------------------------------------+
+| external_flash_mx25v1635f        | The DUT has a mx25v1635f SPI flash connected to the board                                  |
++----------------------------------+--------------------------------------------------------------------------------------------+
+| external_xtal                    | The DUT has an external 32768 Hz crystal                                                   |
++----------------------------------+--------------------------------------------------------------------------------------------+
+| fgpio_loopback                   | The DUT has a FGPIO controller and J1-14 shortend with J1-16                               |
```
Comment (hakehuang) [inline doc/develop/test/twister/hardware-fixtures.rst:19]: the description should be more general, not for one dedicated platform.

File: doc/develop/test/twister/hardware-fixtures.rst [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+| fixture_sdhc                     | The DUT can perform SDHC tests                                                             |
++----------------------------------+--------------------------------------------------------------------------------------------+
+| gpio_loopback                    | The DUT has test specific GPIO pins connected together                                     |
++----------------------------------+--------------------------------------------------------------------------------------------+
+| gpio_spi_loopback                | The DUT has two SPI busses connected together                                              |
++----------------------------------+--------------------------------------------------------------------------------------------+
+| gpio_timerout                    | The DUT can perform external timing measurements on a timer toggled GPIO                   |
++----------------------------------+--------------------------------------------------------------------------------------------+
+| i2c_bus_short                    | The DUT has two I2C busses connected together and the I2C controller are DUAL role capable |
```
Comment (hakehuang) [inline doc/develop/test/twister/hardware-fixtures.rst:29]: I remember there is i2c_bus_target, will missing here?

File: doc/develop/test/twister/hardware-fixtures.rst [hunk 6]
```diff
@@ -0,0 +1,56 @@
+Hardware fixture table
+^^^^^^^^^^^^^^^^^^^^^^
+
++----------------------------------+--------------------------------------------------------------------------------------------+
```
Comment (hakehuang) [inline doc/develop/test/twister/hardware-fixtures.rst:4]: would this be better that we add one column for the testcase name that use this fixture?

**Pull request closed without merging**